packages/blade/src/components/IconButton/StyledIconButton.web.tsx

```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components';
+
+type StyledClickableProps = {
+  contrast: 'low' | 'high';
+};
+
+const StyledClickable = styled.button<StyledClickableProps>((props) => {
+  const { theme, contrast } = props;
+  const linkColorToken = theme.colors.feedback.neutral.action.icon.link; // todo: should be from surfaceAction
+  const contrastToken = contrast === 'high' ? 'highContrast' : 'lowContrast';
+  const motionToken = theme.motion;
+
+  return {
+    border: 'none',
+    cursor: 'pointer',
+    padding: 0,
+    borderRadius: theme.border.radius.small,
+    background: 'transparent',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    color: linkColorToken.default[contrastToken],
+    transition: `all ${motionToken.duration.xquick} ${motionToken.easing.standard.effective}`,
+
+    'svg, path': {
+      // todo: should be ideally set in Icon component
+      fill: 'currentColor',
+    },
+
+    '&:hover': {
+      color: linkColorToken.hover[contrastToken],
+    },
+
+    '&:focus': {
+      outline: 'none',
+      boxShadow: `0px 0px 0px 4px ${theme.colors.brand.primary[400]}`,
+      color: linkColorToken.focus[contrastToken],
+    },
```

Comment:
There are slight deviations here, for example considering button:
- We don't show a border here
- We don't change background

So if we want to abstract only focus ring we can consider moving `outline` and `boxShadow` separately, which does make sense because on Figma this is a separate common abstraction. Will dig around other components and update.

Reply:
Yeah. At the very least have a `getFocusRingStyles()`?

Reply:
I see that we're using focus ring in different ways, in some components as CSS and at some places via helper functions. Probably can be taken up as a separate PR.

Reply:
Tracking https://github.com/razorpay/blade/issues/638